tests/pipeline/test_update_ctx_misc.py

```diff
@@ -33,3 +34,40 @@ def call(self, ctx: Context) -> bool:
     ctx = await pipeline._run_pipeline(Message(), 0, update_ctx_misc={"condition": False})
 
     assert ctx.last_response.text == "failure"
+
+
+@pytest.mark.asyncio
+async def test_context_order():
+
+    class LongResponse(BaseResponse):
```

Comment:
Make one class (instead of `LongResponse` and `LongerResponse`) that retrieves the amount to sleep from `ctx.last_request`.

Then you only have one class, there's no need to add `ExactMatch` conditions and all sleep times can be parametrized via a single point (asyncio.gather of `_run_pipeline`).